features/neuropixel_features.py

```diff
@@ -0,0 +1,141 @@
+from riglib.experiment import traits
+import requests
+import json
+from riglib import experiment
+
+# This class is almost same as open ephys python tool other than timeout option
+class OpenEphysHTTPServer:
+    def __init__(self, address='10.155.205.108',timeout=0.5):
+        self.address = 'http://' + address + ':37497'
+        self.timeout = timeout
+
+    def send(self, endpoint, payload=None):
+        # Send a request to the server
+        try:
+            if payload is None:
+                resp = requests.get(self.address + endpoint, timeout=self.timeout)
+            else:
+                resp = requests.put(self.address + endpoint, data = json.dumps(payload), timeout=self.timeout)
+            response = resp.json()
+        except Exception as e:
+            print(e, 'Check if you started openephys in neuropixel computer')
+            resp = {}
+            resp['mode'] = 'No connection with openephys'
+            response = resp
+        return response
+
+    def set_parent_dir(self,path):
+        # Set the parent directory
+        payload = {'parent_directory' : path}
+        data = self.send('/api/recording', payload=payload)
+        return data
+
+    def set_prepend_text(self,text):
+        # Set the prepend name for the recording directory
+        payload = {'prepend_text' : text}
+        data = self.send('/api/recording', payload=payload)
+        return data
+
+    def set_base_text(self,text):
+        # Set the base name for the recording directory
+        payload = {'base_text' : text}
+        data = self.send('/api/recording', payload=payload)
+        return data
+
+    def set_append_text(self,text):
+        # Set the append name for the recording directory
+        payload = {'append_text' : text}
+        data = self.send('/api/recording', payload=payload)
+        return data
+
+    def status(self):
+        # Get the current status of the GUI (IDLE, ACQUIRE, or RECORD)
+        return self.send('/api/status')['mode']
+
+    def acquire(self):
+        # Start acquisition
+        payload = { 'mode' : 'ACQUIRE',}
+        data = self.send('/api/status', payload=payload)
+        return data['mode']
+
+    def record(self):
+        # Start recording data
+        previous_mode = self.status()
+        payload = {'mode' : 'RECORD',}
+        data = self.send('/api/status', payload=payload)       
+        return data['mode']
+
+    def idle(self):
+        # Stop acquisition or recording data
+        previous_mode = self.status()
+        payload = { 'mode' : 'IDLE',}
+        data = self.send('/api/status', payload=payload)
+        return data['mode']
+
+
+class RecordNeuropixels(traits.HasTraits):
+
+    neuropixel_port1 = traits.Int(0, desc="Channel number you used for recoding with port 1 probe")
+    neuropxiel_port2 = traits.Int(0, desc="Channel number hole you used for recoding with port 2 probe")
+    neuropixel_mapping_file = traits.String("", desc="Name of channel mapping file")
```

Comment:
maybe also want to add parameters: 
- `neuropixel_port1_chamber` and `neuropixel_port2_chamber` which would be "LM1" for now but might change for future implants
- `neuropixel_portX_drive_type` which would indicate the name of the hole layout you're using
- `neuropixel_portX_site`,  which could be the number of the hole you penetrate through
- `neuropixel_portX_depth`, which would be from the narishige drive